### DISCUSSION TITLE
[SUPPORT] Assign read permission to the Bitcoin group users for electrs

### DISCUSSION THREAD

**Discussion by smartm0use** [minibolt-guide/minibolt#84] [Ideas]
electrs doesn't launch at startup using [systemd service](https://v2.minibolt.info/bonus-guides/bitcoin/electrs#create-systemd-service).

To start it I have first to run `chmod g+r /home/bitcoin/.bitcoin/.cookie` and then `sudo systemctl start electrs`.

For bitcoin-core you fixed the same issue with `startupnotify=chmod g+r /home/bitcoin/.bitcoin/.cookie` row in bitcoin.conf file.

How to reach the same for electrs?

**Comment by twofaktor** [comment]
I don't understand you, bitcoind creates the .cookie file and assigns read permission at startup for the group members can read it (i.e electrs) what are you suggesting?

**Comment by smartm0use** [comment]
I'm trying to figure out what's the issue. On every startup electrs doesn't start automatically and when I run it manually I get an error related to `.cookie` file. Then if i run `chmod g+r /home/bitcoin/.bitcoin/.cookie` and then run again electrs it works.

User `electrs` already belong to `bitcoin` group:

```
electrs@<my-machine>:~$ groups
electrs bitcoin
```

What could be the issue in my case?

**Comment by twofaktor** [comment]
> I'm trying to figure out what's the issue. On every startup electrs doesn't start automatically and when I run it manually I get an error related to `.cookie` file. Then if i run `chmod g+r /home/bitcoin/.bitcoin/.cookie` and then run again electrs it works.
> 
> User `electrs` already belong to `bitcoin` group:
> 
> ```
> electrs@<my-machine>:~$ groups
> electrs bitcoin
> ```
> 
> What could be the issue in my case?

I don't know, but shouldn't be like that, `startupnotify=chmod g+r /home/bitcoin/.bitcoin/.cookie` is defined on `bitcoin.conf` to automatically assign the permission at startup Bitcoin Core https://v2.minibolt.info/bitcoin/bitcoin/bitcoin-client#configuration

![imagen](https://github.com/minibolt-guide/minibolt/assets/89636253/458b0acb-b242-4bba-a507-338af0824bf3)

**Comment by smartm0use** [comment]
Yes, it the configuration line I wrote in my first message... that doens't work with electrs.
Weird...

**Comment by st3b1t** [comment]
I have similar behavior... In my node v24 startupnotify command is executed after about 10/15 seconds after bitcoin service is started.
.cookie file is created immediately but... the chmod applied after some seconds..
you can monitor it with:
```bash
/home/bitcoin# watch -n1 ls -l .bitcoin/.cookie
```

**Comment by twofaktor** [comment]
I think is possible to sleep electrs for seconds before startup, to give time to bitcoind start, but I'm not sure how do it

**Comment by st3b1t** [comment]
@twofaktor I created this script to check permissions:
https://gist.github.com/st3b1t/e99d7a8f7a55b221a3d69da881318d5b

could be added in the docker command or in the Electrs startup script

**Comment by twofaktor** [comment]
> @twofaktor I created this script to check permissions: https://gist.github.com/st3b1t/e99d7a8f7a55b221a3d69da881318d5b
> 
> could be added in the docker command or in the Electrs startup script

I think we should find another solution, docker and scripts are not allowed on RaspiBolt/MiniBolt guides

I think should be possible to use some service parameters or maybe the Electrs dev include this fix directly on the code (ideal), which could be requested opening an issue https://github.com/romanz/electrs/issues

**Comment by hellresistor** [comment]
friends. correction on line on .conf file..

bitcoind = 127.0.0.1:8332
rpccookie = /data/bitcoin/.cookie

NOT /home/....

**Reply by twofaktor** [reply]
What do you mean? That is exactly the same

**Reply by hellresistor** [reply]
on manual say "rpccookie = /home/bitcoin/.cookie"
but should be "rpccookie = /data/bitcoin/.cookie"

switch directory is not same ^^

this applies to electrs and fulcrum

**Reply by twofaktor** [reply]
on manual say "rpccookie = /data/bitcoin/.cookie" not "rpccookie = /home/bitcoin/.cookie"

![imagen](https://github.com/minibolt-guide/minibolt/assets/89636253/60273403-c79d-4e67-9e5f-f737b0677432)

**Reply by hellresistor** [reply]
ooh. and i had made this change. removed from bitcoind.service
# /home/bitcoin/.bitcoin/bitcoin.conf
startupnotify=chmod g+r /data/bitcoin/.cookie 

I just telling a possible way to solve o/ (at least for me)

**Reply by hellresistor** [reply]
now i understand how i arrived here xD
someone mention minibolt issue xDD

https://github.com/raspibolt/raspibolt/issues/1375#event-10948047249
